From e17648da3e90b4bec9ca4eebea7c9ddeed07fd9d Mon Sep 17 00:00:00 2001
From: Igor Grinberg <grinberg@compulab.co.il>
Date: Sun, 24 Feb 2013 19:07:56 +0200
Subject: ARM: i.MX6: cm-fx6: refactor I2C3 pins handling

I2C3 pins on cm-fx6 are not designed to be changed in run time and are
always belong to the I2C3 bus.
Refactor the I2C3 pins to simplify the code.

Signed-off-by: Igor Grinberg <grinberg@compulab.co.il>
---
 arch/arm/mach-mx6/board-cm-fx6-dl.h |    9 ++++-----
 arch/arm/mach-mx6/board-cm-fx6-q.h  |    9 ++++-----
 arch/arm/mach-mx6/board-cm-fx6.c    |   19 ++++---------------
 3 files changed, 12 insertions(+), 25 deletions(-)

diff --git a/arch/arm/mach-mx6/board-cm-fx6-dl.h b/arch/arm/mach-mx6/board-cm-fx6-dl.h
index 00eb1b9..bc197e8 100644
--- a/arch/arm/mach-mx6/board-cm-fx6-dl.h
+++ b/arch/arm/mach-mx6/board-cm-fx6-dl.h
@@ -90,6 +90,10 @@ static iomux_v3_cfg_t mx6dl_arm2_pads[] = {
 	MX6DL_PAD_KEY_COL3__I2C2_SCL,
 	MX6DL_PAD_KEY_ROW3__I2C2_SDA,
 
+	/* I2C3 */
+	MX6DL_PAD_GPIO_3__I2C3_SCL,
+	MX6DL_PAD_GPIO_6__I2C3_SDA,
+
 	/* DISPLAY */
 	MX6DL_PAD_DI0_DISP_CLK__IPU1_DI0_DISP_CLK,
 	MX6DL_PAD_DI0_PIN15__IPU1_DI0_PIN15,
@@ -136,11 +140,6 @@ static iomux_v3_cfg_t mx6dl_arm2_pads[] = {
 	MX6DL_PAD_GPIO_1__USBOTG_ID,
 };
 
-static iomux_v3_cfg_t mx6dl_arm2_i2c3_pads[] = {
-	MX6DL_PAD_GPIO_3__I2C3_SCL,
-	MX6DL_PAD_GPIO_6__I2C3_SDA,
-};
-
 static iomux_v3_cfg_t mx6dl_arm2_spdif_pads[] = {
 	/* SPDIF */
 	MX6DL_PAD_GPIO_16__SPDIF_IN1,
diff --git a/arch/arm/mach-mx6/board-cm-fx6-q.h b/arch/arm/mach-mx6/board-cm-fx6-q.h
index 727e6e1..b46a25f 100644
--- a/arch/arm/mach-mx6/board-cm-fx6-q.h
+++ b/arch/arm/mach-mx6/board-cm-fx6-q.h
@@ -90,6 +90,10 @@ static iomux_v3_cfg_t mx6q_arm2_pads[] = {
 	MX6Q_PAD_KEY_COL3__I2C2_SCL,
 	MX6Q_PAD_KEY_ROW3__I2C2_SDA,
 
+	/* I2C3 */
+	MX6Q_PAD_GPIO_3__I2C3_SCL,
+	MX6Q_PAD_GPIO_6__I2C3_SDA,
+
 	/* DISPLAY */
 	MX6Q_PAD_DI0_DISP_CLK__IPU1_DI0_DISP_CLK,
 	MX6Q_PAD_DI0_PIN15__IPU1_DI0_PIN15,
@@ -136,11 +140,6 @@ static iomux_v3_cfg_t mx6q_arm2_pads[] = {
 	MX6Q_PAD_GPIO_1__USBOTG_ID,
 };
 
-static iomux_v3_cfg_t mx6q_arm2_i2c3_pads[] = {
-	MX6Q_PAD_GPIO_3__I2C3_SCL,
-	MX6Q_PAD_GPIO_6__I2C3_SDA,
-};
-
 static iomux_v3_cfg_t mx6q_arm2_spdif_pads[] = {
 	/* SPDIF */
 	MX6Q_PAD_GPIO_16__SPDIF_IN1,
diff --git a/arch/arm/mach-mx6/board-cm-fx6.c b/arch/arm/mach-mx6/board-cm-fx6.c
index bc4ac7f..36a7287 100644
--- a/arch/arm/mach-mx6/board-cm-fx6.c
+++ b/arch/arm/mach-mx6/board-cm-fx6.c
@@ -799,12 +799,10 @@ static void __init cm_fx6_init(void)
 	iomux_v3_cfg_t *common_pads = NULL;
 	iomux_v3_cfg_t *spdif_pads = NULL;
 	iomux_v3_cfg_t *flexcan_pads = NULL;
-	iomux_v3_cfg_t *i2c3_pads = NULL;
 
 	int common_pads_cnt;
 	int spdif_pads_cnt;
 	int flexcan_pads_cnt;
-	int i2c3_pads_cnt;
 
 
 	/*
@@ -816,41 +814,32 @@ static void __init cm_fx6_init(void)
 		common_pads = mx6q_arm2_pads;
 		spdif_pads = mx6q_arm2_spdif_pads;
 		flexcan_pads = mx6q_arm2_can_pads;
-		i2c3_pads = mx6q_arm2_i2c3_pads;
 
 		common_pads_cnt = ARRAY_SIZE(mx6q_arm2_pads);
 		spdif_pads_cnt =  ARRAY_SIZE(mx6q_arm2_spdif_pads);
 		flexcan_pads_cnt = ARRAY_SIZE(mx6q_arm2_can_pads);
-		i2c3_pads_cnt = ARRAY_SIZE(mx6q_arm2_i2c3_pads);
 	} else if (cpu_is_mx6dl()) {
 		common_pads = mx6dl_arm2_pads;
 		spdif_pads = mx6dl_arm2_spdif_pads;
 		flexcan_pads = mx6dl_arm2_can_pads;
-		i2c3_pads = mx6dl_arm2_i2c3_pads;
 
 		common_pads_cnt = ARRAY_SIZE(mx6dl_arm2_pads);
 		spdif_pads_cnt =  ARRAY_SIZE(mx6dl_arm2_spdif_pads);
 		flexcan_pads_cnt = ARRAY_SIZE(mx6dl_arm2_can_pads);
-		i2c3_pads_cnt = ARRAY_SIZE(mx6dl_arm2_i2c3_pads);
 	}
 
 	BUG_ON(!common_pads);
 	mxc_iomux_v3_setup_multiple_pads(common_pads, common_pads_cnt);
 
-	/*
-	 * IEEE-1588 ts_clk, S/PDIF in and i2c3 are mutually exclusive
-	 * because all of them use GPIO_16.
-	 * S/PDIF out and can1 stby are mutually exclusive because both
-	 * use GPIO_17.
-	 */
 	if (spdif_en) {
 		BUG_ON(!spdif_pads);
 		mxc_iomux_v3_setup_multiple_pads(spdif_pads, spdif_pads_cnt);
-	} else {
-		BUG_ON(!i2c3_pads);
-		mxc_iomux_v3_setup_multiple_pads(i2c3_pads, i2c3_pads_cnt);
 	}
 
+	/*
+	 * S/PDIF out and can1 stby are mutually exclusive because both
+	 * use GPIO_17.
+	 */
 	if (!spdif_en && flexcan_en) {
 		BUG_ON(!flexcan_pads);
 		mxc_iomux_v3_setup_multiple_pads(flexcan_pads,
-- 
1.7.9.5

